packages/react-avatar/src/components/AvatarGroup/AvatarGroup.types.ts

```diff
@@ -1,19 +1,72 @@
 import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';
+import { PopoverSurface } from '@fluentui/react-popover';
+import { AvatarSizes } from '../Avatar/Avatar.types';
+import { Button } from '@fluentui/react-button';
+import { TooltipProps } from '@fluentui/react-tooltip';
 
 export type AvatarGroupSlots = {
   root: Slot<'div'>;
+
+  /**
+   * Popover trigger slot that can be used to change the overflow indicator.
+   */
+  popoverTrigger?: Slot<typeof Button>;
+
+  /**
+   * Popover surface that will be displayed when the popover is triggered.
+   */
+  popoverSurface?: Slot<typeof PopoverSurface>;
 };
 
 type AvatarGroupCommons = {
-  // TODO Add things shared between props and state here
+  /**
+   * Layout the Avatars should be displayed as.
+   * @default grid
+   */
+  layout: 'grid' | 'stacked' | 'pie';
+
+  /**
+   * Maximum number of Avatars to be displayed before overflowing.
+   * NOTE: if pie layout is used, `maxAvatars` will be overwritten.
+   * @default 5
+   */
+  maxAvatars: number;
+
+  /**
+   * Size of the avatars.
+   * @default 32
+   */
+  size: AvatarSizes;
+
+  /**
+   * Wheter the overflow indicator should render an icon instead of the number of overflowed avatars.
+   * @default false
+   */
+  iconOverflowIndicator: boolean;
```

Comment:
My feeling is that this prop isn't naturally a boolean. While it's clear that a `true` value means "show an icon", it's not clear what a `false` value means. Plus, it's possible we may want to add other overflow options later.

What do you think about?
```suggestion
  overflowIndicator: 'number-overflowed' | 'icon';
```

Default value will be `number-overflowed`.